### PR TITLE
Add more tables on using 2gamma,and add add tables on using gamma.

### PR DESCRIPTION
**Pull request by sophy1029**
In the 2gamma case, I add results of the test example on Q3 with gamma = 1, on Q4 with gamma = 1, on Q3 with gamma = 2, and Q4 with gamma = 2. From the results, we can see that smaller penalties don't work well on Q2 elements, but if we test it on higher orders, like Q4, the results converge as theoretical expectations. 

In the gamma case, I add tables of all conditions.

### CODE REVIEW

**Comment by bangerth** [comment]
That's too many tables :-) Remember that there is a difference between *data* and *information*. You are showing a lot of data, and mostly leave it to the reader to sort through it and make sense of it. I think a better approach would be to say something along the lines of "It is easy to repeat this experiment but with gamma=... and see that ...". In other words, I would much rather replace the additional tables by text that *explains*, unless a table shows something truly unexpected.

File: examples/step-71/doc/results.dox [hunk 1]
```diff
@@ -141,6 +141,48 @@ follow the theoretical expectations, the $H^2$-seminorm does not seem to converg
...
+   <td>   3                  </td><td>    4.057e-03   </td><td>  1.99  </td><td> 1.127e-01  </td><td>   1.01    </td><td> 3.179  </td><td> 0.08</td>
+  </tr>
+  <tr>
+   <td>   4                  </td><td>   1.033e-03   </td><td> 1.97   </td><td> 5.726e-02    </td><td> 0.97       </td><td>  3.22 </td><td> -0.01 </td>
+  </tr>
+  <tr>
+   <td>   5                  </td><td>   2.551e-04 </td><td>  2.01  </td><td>  2.831e-02 </td><td>   1.01    </td><td> 3.142 </td><td> 0.03   </td>
+  </tr>
+</table>
+This does not converge as theoretical expectations.
```
Comment (bangerth) [inline examples/step-71/doc/results.dox:163]: For example, you could replace the whole section you introduce here by just saying something like

  Given that gamma=1 is already too small for Q_2 elements, it may not be surprising that if one repeated the
  experiment with a Q_3 element, the results are even more disappointing: One again only obtains convergence
  rates of 2, 1, zero -- i.e., no better than for the Q_2 element (although the errors are smaller in magnitude).
  Maybe surprisingly, however, one obtains more or less the expected convergence orders when using Q_4
  elements. Regardless, this uncertainty suggests that gamma=1 is at best a risky choice, and at worst an
  unreliable one and that we should choose gamma larger.

File: examples/step-71/doc/results.dox [hunk 2]
```diff
@@ -172,6 +214,53 @@ is already too small even for the value $p=2$ chosen here, and one can
...
+  </tr>
+  <tr>
+   <td>   4                  </td><td>  1.713e-05   </td><td>  3.92   </td><td> 1.078e-03    </td><td> 3.05      </td><td> 9.365e-02 </td><td> 2.06 </td>
+  </tr>
+  <tr>
+   <td>   5                  </td><td>   1.095e-06  </td><td> 3.96  </td><td> 1.348e-04 </td><td>    3.00  </td><td> 2.337e-02  </td><td> 2.00  </td>
+  </tr>
+</table>
+In this case, the convergence rates follow the
+theoretical expectations.
```
Comment (bangerth) [inline examples/step-71/doc/results.dox:238]: The same is true here. I would replace this and the following table by text of the following kind:

  Again, we could repeat this kind of experiment for Q_3 and Q_4 elements. In both cases, we will find that we
  obtain roughly the expected convergence rates. Of more interest may then be to compare the absolute
  size of the errors. While in the table above, for the Q_2 case, the errors on the finest grid are comparable between
  the gamma=p(p+1) and gamma=2 case, for Q_3 the errors are substantially larger for gamma=2 than for 
  gamma=p(p+1). The same is true for the Q_4 case.

File: examples/step-71/doc/results.dox [hunk 3]
```diff
@@ -201,3 +290,220 @@ make sense:
...
+    <tr>
+     <td>   4                  </td><td>   1.103e-03   </td><td>  1.67   </td><td> 6.106e-03    </td><td>  1.83        </td><td>   4.015e-01 </td><td> 1.016  </td>
+    </tr>
+    <tr>
+     <td>   5                  </td><td>  3.084e-04  </td><td>  1.83   </td><td>  1.622e-03   </td><td>    1.91        </td><td> 1.993e-01 </td><td>  1.010   </td>
+    </tr>
+  </table>
+  We can see in this case, convergence rates satisfy theoretical expectations.
+  Comparing taking $\gamma$ and taking $2\gamma$ in the `boundary_worker`,
+  $2\gamma$ converges better.
```
Comment (bangerth) [inline examples/step-71/doc/results.dox:322]: How do you determine this? I'm seeing the following errors for the original case on the finest mesh:
  3.277e-4   1.666e-3   1.981e-1
and you have here
  3.084e-4   1.622e-3   1.993e-1
So they are almost the same.

File: examples/step-71/doc/results.dox [hunk 4]
```diff
@@ -201,3 +290,220 @@ make sense:
...
+     <td>   4                  </td><td>   9.404e-09  </td><td> 4.83   </td><td> 1.040e-06    </td><td> 3.91       </td><td> 1.774e-04 </td><td> 3.07 </td>
+    </tr>
+    <tr>
+     <td>   5                  </td><td>   7.943e-10 </td><td>  3.56  </td><td>   6.693e-08 </td><td> 3.95     </td><td> 2.150e-05  </td><td> 3.04    </td>
+    </tr>
+  </table>
+  We can see that the $L_2$ norm convergence rates are around 5,
+  $H^1$-seminorm convergence rates are around 4,
+  and $H^2$-seminorm convergence rates are around 3.
+  The $L_2$ norm convergence rate in this case is the same as taking $2\gamma$ previously.
```
Comment (bangerth) [inline examples/step-71/doc/results.dox:372]: Yes, so again the results are very much comparable. It is not necessary to spend a page of tables on *data* that has little actual *information*.

**Comment by bangerth** [comment]
So having looked through the data you show, there seems to be little difference between the gamma/2*gamma case that's currently implemented, and the gamma/gamma case you tested. It's often useful to do the sort of experiment you made to convince yourself that there is little difference, but in most cases my suggestion would then be to just change the code rather than add several pages of tables.

In other words:
* Just change the code to use the gamma/gamma case
* Just update the existing tables so that they reflect the state of the code; you already have these tables created, you just need to copy them to the places where we currently have them.

**Comment by sophy1029** [comment]
Thanks, Dr. Bangerth. Yes, it's much better to use text to explain rather
than showing all the data.

So we will choose the case which has gamma in face_worker,  gamma in
boundary_worker, and gamma in cell-rhs, rather than 2gamma?
If so, I will update the step-71.cc file.

In the results section, shall we still consider totally 5 cases:
gamma = k(k+1) with Q2, gamma = k(k+1) with Q3, gamma = k(k+1) with Q4,
gamma = 1 with Q2, and gamma = 2 with Q2?
And we can mention smaller penalties are risky choices, like you explained,
rather than showing too many tables.

**Comment by bangerth** [comment]
On 12/30/19 12:06 PM, Zhuoran Wang wrote:
> 
> So we will choose the case which has gamma in face_worker, gamma in
> boundary_worker, and gamma in cell-rhs, rather than 2gamma?
> If so, I will update the step-71.cc file.

Yes, please!


> In the results section, shall we still consider totally 5 cases:
> gamma = k(k+1) with Q2, gamma = k(k+1) with Q3, gamma = k(k+1) with Q4,
> gamma = 1 with Q2, and gamma = 2 with Q2?
> And we can mention smaller penalties are risky choices, like you explained,
> rather than showing too many tables.

Yes, this sounds like the most reasonable approach.

**Comment by sophy1029** [comment]
Thanks for you quick response!
I will update these as soon as possible.
I’m on the flight to China, I will update when I get home in China.

**Comment by bangerth** [comment]
On 12/30/19 12:35 PM, Zhuoran Wang wrote:
> Thanks for you quick response!
> I will update these as soon as possible.
> I’m on the flight to China, I will update when I get home in China.

Ha, no worries -- it's not urgent :-)
